pkg/bytecode/vm_test.go

```diff
@@ -595,6 +595,22 @@ func TestIf(t *testing.T) {
 			x = x`,
 			wantStackTop: makeValue(t, 1),
 		},
+		{
+			name: "local scope",
+			input: `x := 1
+			if false
+				x := 2
+				x = x + 1
+			else if false
+				x := 3
+				x = x + 1
+			else if false
+				x := 4
+				x = x + 1
+			end
+			x = x`,
```

Comment:
How about a test like:

```
x := 3
if true
   x := 2
   x = x
end
if x == 3
   // this branch should be taken, 
   // we think it isn't because shadowing isn't implemented:
    x = 1 
end
// ensure x is 1 NOT 2
```

Reply:
There is an existing test that covers shadowing, so I'll just leave that.